test/Datadog.Trace.Tests/RuntimeMetrics/RuntimeMetricsWriterTests.cs

```diff
@@ -88,7 +88,11 @@ public void ShouldCaptureFirstChanceExceptions()
                 writer.PushEvents();
 
                 statsd.Verify(
-                    s => s.Increment(MetricsNames.ExceptionsCount, It.IsAny<int>(), It.IsAny<double>(), It.IsAny<string[]>()),
+                    s => s.Increment(MetricsNames.ExceptionsCount, It.IsAny<int>(), It.IsAny<double>(), new[] { "exception_type:CustomException1" }),
```

Comment:
Should we add this to the other "never" verification in lines 71-73 above, too?

Reply:
Yeah good point, will do

Reply:
Done in 480242c4869d94210048d348978ff35011f08ce1

Reply:
Actually, in retrospect, that first one is not needed, because we don't push ANY exceptions until the PushEvents method is called. I'm actually going to revert that last commit

Reply:
In the PR description you said "other threads in the same AppDomain could be throwing exceptions which our runtime metrics would catch." That doesn't apply to this check?

Reply:
The very first one, it does not, because we don't submit any metrics to StatsD instance until the first `writer.PushEvents()`. That's why it's good to keep the broad `It.IsAny<string[]>()` check as it is before the first `writer.PushEvents()` call

Reply:
My _theory_ is that there is a race condition in between the `statsd.Invocations.Clear()` and the final `writer.PushEvents()` where exceptions thrown from other test methods could be collected by this runtime metrics writer. We're really just trying to check that the local exceptions we threw got collected, written to StatsD, and will not show up again. For that reason, I'm proposing to only check the custom exceptions

Reply:
Sounds good. Thanks for clarifying. I will resolve this comment.